core/src/main/java/org/jdbi/v3/core/Handle.java

```diff
@@ -51,166 +40,50 @@
  * This represents a connection to the database system. It usually is a wrapper around
  * a JDBC Connection object.
  */
-public class Handle implements Closeable
+public class Handle implements Closeable, Configurable<Handle>
 {
     private static final Logger LOG = LoggerFactory.getLogger(Handle.class);
 
-    protected final JdbiConfig config;
-    protected StatementBuilder statementBuilder;
-    private boolean closed = false;
-    protected final TransactionHandler transactions;
-    protected final Connection connection;
+    private final TransactionHandler transactions;
+    private final Connection connection;
 
-    private ThreadLocal<ExtensionMethod> extensionMethod = new ThreadLocal<>();
+    private ThreadLocal<ConfigRegistry> config;
```

Comment:
I think this is worthy of mentioning in a comment.  I probably would have figured this out eventually but it's at least a little subtle :)